### ISSUE TITLE
Use comment shadow table data for serializing users' comments

### ISSUE THREAD

**Issue by gsidebo** [mitodl/open-discussions#1712]
https://github.com/mitodl/open-discussions/pull/1603 added users' posts and comments to the profile page. Unfortunately, due to PRAW's lazy-loading-only approach, serializing a user's comments involves 2 expensive HTTP requests to reddit _each_. As a result, it takes quite a long time to load a user's comment feed. To avoid HTTP requests to reddit for these comments, we should do something similar to `PostProxy` (https://github.com/mitodl/open-discussions/commit/8a3e35101b9ea7e3a4a54f67901fa8a836bd6f29), and enable the comment serializer to work with those proxy objects.

**Comment by pdpinch** [comment]
@rhysyngsun was this addressed in #1782?

**Comment by rhysyngsun** [comment]
@pdpinch no, this is related to our user-facing APIs and #1782 and more broadly #1662 are search-impacting only.
